File: app/src/main/java/io/github/wulkanowy/di/AppComponent.java
package io.github.wulkanowy.di;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;
import io.github.wulkanowy.WulkanowyApp;

@Singleton
@Component(modules = {
        AndroidSupportInjectionModule.class,
        AppModule.class,
        BuilderModule.class
})
public interface AppComponent extends AndroidInjector<WulkanowyApp> {
    @Component.Builder
    abstract class Builder extends AndroidInjector.Builder<WulkanowyApp> {
    }
}

File: app/src/main/java/io/github/wulkanowy/di/AppComponent.kt
package io.github.wulkanowy.di

import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import io.github.wulkanowy.WulkanowyApp
import io.github.wulkanowy.data.RepositoryModule
import io.github.wulkanowy.services.ServicesModule
import javax.inject.Singleton

@Singleton
@Component(modules = [
    AndroidSupportInjectionModule::class,
    AppModule::class,
    RepositoryModule::class,
    ServicesModule::class,
    BuilderModule::class])
interface AppComponent : AndroidInjector<WulkanowyApp> {

    @Component.Factory
    interface Factory : AndroidInjector.Factory<WulkanowyApp>
}
